### DISCUSSION TITLE
Managing Actions storage space

### DISCUSSION THREAD

**Discussion by thisismydesign** [community/community#26542] [Actions]
<p>On <a href="https://github.com/organizations/my-org/settings/billing" rel="nofollow noopener">https://github.com/organizations/my-org/settings/billing</a> I can see that our organization is using 2.5 GBs (out of which 2 would be free). I also see these warnings:</p>
<p>&gt; You’ve used 100% of included services for GitHub Storage (GitHub Actions and Packages)</p>
<p>&gt; GitHub Actions and Packages won’t work until a monthly spending limit is set.</p>
<p>Nevertheless, Actions still fully works.</p>
<p>It is not clear what counts towards this usage, what this restriction will affect and how to mitigate it.</p>
<p>We extensively use caching and artifacts features. I’m assuming that either or both count towards the limit. We especially have large (~100MB) artifacts for a certain job (my guess for the culprit). This would result in an infinitely growing storage size.</p>
<p>I have a few observations / possibly feature requests:</p>
<ul>
<li>
<p>I don’t see a way to delete artifacts, nor would I want to do it manually. Having old artifacts which don’t fit the quota removed automatically is reasonable, as it is being done for the cache feature (“The total size of all caches in a repository don’t exceed 2 GB. If you exceed this limit, GitHub will save your cache but will begin evicting caches until the total size is less than 2 GB.”).</p>
</li>
<li>
<p>I think features recycling storage space (e.g. cache) should not count towards the limit, or the sum of them should fit well within the limit. For example, if I currently use the cache feature and it counts towards the limit it can just fill up automatically and I won’t be able to use other features.</p>
</li>
<li>
<p>If artifacts were to be recycled, I would like to be able to keep some of them indefinitely (understandably these would count towards the limit). For example, I would be OK with recycling dev build artifacts but not release build artifacts.</p>
</li>
<li>
<p>For recycled items, I would like to be able to set a custom limit per repository. Since the limit is shared within the organization and some repositories may use it more extensively than others it’s possible that a single project takes up all of the limit. I would like to avoid this.</p>
</li>
</ul>

**Comment by thisismydesign** [comment]
<p>After reaching out to support I got the following response:</p>
<blockquote>
<p>At this time, artifacts are deleted automatically on a rolling 90-day basis. Our team is working to add the ability to delete artifacts, but with the holidays approaching it will likely be January before that feature is available. The initial rollout will likely be a very basic version that leaves room for iteration to perhaps allow customization at a later point.</p>
</blockquote>

**Comment by thisismydesign** [comment]
<p>It is now possible to delete artifacts manually.</p>
<blockquote>
<p>To delete artifacts, visit a build and click on the Artifacts link in the upper right. If you have write access to the repo, you should see red trashcan icons to remove any artifacts you no longer need.</p>
</blockquote>

**Comment by geoperez** [comment]
<p>Do you know if there is a setting to change this value?</p>

**Comment by jamesdh** [comment]
<p>“Visit a build”? What is meant by this? The term build isn’t really used anywhere in GitHubs vocabulary. </p>
<p>Do you mean the result of a GitHub Action job? Or a built package in the Packages repo? In either case, I don’t see anything related to artifacts or removal. </p>

**Comment by thisismydesign** [comment, marked as answer]
<p>Our team has created an Action to periodically clean up old artifacts:</p>
<p><a href="https://github.com/c-hive/gha-remove-artifacts" rel="nofollow noopener">https://github.com/c-hive/gha-remove-artifacts</a></p>
<p>It also supports keeping tagged (release) artifacts.</p>

**Reply by tigranqless** [reply]
@thisismydesign did this solution work for you?

**Comment by thisismydesign** [comment]
<p>There isn’t one at the moment but we created an action to solve this, see the accepted solution: <a href="https://github.community/t/managing-actions-storage-space/16973/6">https://github.community/t5/GitHub-Actions/Managing-Actions-storage-space/m-p/51291#M8134</a></p>

**Comment by thisismydesign** [comment]
<p>Actions -&gt; Select a run in the list -&gt; Select a job on the left side</p>
<p>You end up at e.g. <a href="https://github.com/org/repo/runs/533166375" rel="nofollow noopener">https://github.com/org/repo/runs/533166375</a></p>
<p>In the upper-right corner, you will see an <code>Artifacts</code> menu. After clicking you will see the artifacts and can delete them as well.</p>
<p><img src="https://user-images.githubusercontent.com/3299948/181098407-b254740c-91d6-4539-aa1f-28e10d58ea62.png" alt="Screenshot from 2020-03-25 13-34-36.png" data-base62-sha1="vlAz41KFY5znTwp9C5CSMdZSygW" width="427" height="149"></p>

**Comment by Alessthana** [comment]
<p>Hey! What do you mean by old artifacts? The one before the previous builds artifacts and releases right?</p>

**Comment by varunvora** [comment]
<p>Is there a way we can find out which artifacts are using the most space?</p>

**Comment by Rebel028** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/5/45d7c9599fc248ca8720bc65717d2221d48c912f.png" data-download-href="/uploads/short-url/9XRgukLL97iXQSesHCX3jSiaxdR.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/27502053/181098408-d31f5c7f-f95c-4b8c-a03c-ecff194eb860.png" alt="image" data-base62-sha1="9XRgukLL97iXQSesHCX3jSiaxdR" width="690" height="120" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/5/45d7c9599fc248ca8720bc65717d2221d48c912f_2_690x120.png, https://aws1.discourse-cdn.com/github/original/3X/4/5/45d7c9599fc248ca8720bc65717d2221d48c912f.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/5/45d7c9599fc248ca8720bc65717d2221d48c912f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/5/45d7c9599fc248ca8720bc65717d2221d48c912f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">967×169 7.05 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You can use “Get usage report” option in billing section, it shows daily usage stats for each repo you have including storage usage</p>

**Comment by olivluca** [comment]
I just got a warning that I reached 75% of github storage for actions and packages.
What troubles me is the phrase **Removing stored artifacts will not reduce this number, but it will lower its rate of growth.**
Does it mean that, no matter what old workflow runs I delete, I will never recover the space?

**Reply by ktryniszewski-mdsol** [reply]
most likely, given how storage space is calculated
https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#sample-storage-cost-calculation

**Reply by olivluca** [reply]
I just checked and my storage usage is 0 now, so maybe the phrase **Removing stored artifacts will not reduce this number, but it will lower its rate of growth** should be interpreted in the context of the current period.
